File: src/main/java/com/minmax/ultradex/jei/PixelmonDrop/PokemonSpawnHelper.java
package com.minmax.ultradex.jei.PixelmonDrop;

import com.pixelmonmod.pixelmon.api.pokemon.Pokemon;
import com.pixelmonmod.pixelmon.api.spawning.SpawnSet;
import com.pixelmonmod.pixelmon.api.spawning.archetypes.entities.pokemon.SpawnInfoPokemon;
import com.pixelmonmod.pixelmon.spawning.PixelmonSpawning;
import net.minecraft.util.ResourceLocation;

import javax.annotation.Nonnull;
import java.util.Collections;
import java.util.List;
import java.util.Map;
import java.util.stream.Collectors;

public class PokemonSpawnHelper {
    public static List<String> getBiomes(Pokemon pokemon) {
        Map<String, List<SpawnSet>> spawnSets = PixelmonSpawning.getAll();
        if (!spawnSets.containsKey("standard")) {
            return null;
        }

        return spawnSets.get("standard")
                .stream()
                .flatMap(spawnSet -> spawnSet.spawnInfos.stream())
                .filter(spawnInfo -> (spawnInfo instanceof SpawnInfoPokemon) && ((SpawnInfoPokemon) spawnInfo).getPokemonSpec().matches(pokemon))
                .flatMap(spawnInfo -> spawnInfo.condition.biomes.stream())
                .map(ResourceLocation::toString)
                .collect(Collectors.toList());
    }
}

File: src/main/kotlin/com/minmax/ultradex/pokemon/PokemonSpawnHelper.kt
package com.minmax.ultradex.pokemon

import com.pixelmonmod.pixelmon.api.pokemon.Pokemon
import com.pixelmonmod.pixelmon.api.spawning.SpawnInfo
import com.pixelmonmod.pixelmon.api.spawning.SpawnSet
import com.pixelmonmod.pixelmon.api.spawning.archetypes.entities.pokemon.SpawnInfoPokemon
import com.pixelmonmod.pixelmon.spawning.PixelmonSpawning
import net.minecraft.util.ResourceLocation
import java.util.*
import java.util.stream.Collectors

object PokemonSpawnHelper {
    @JvmStatic
    fun getBiomes(pokemon: Pokemon): List<String> {
        val spawnSets = PixelmonSpawning.getAll()
        val standardSpawnSet = spawnSets["standard"] ?: return Collections.emptyList()

        return standardSpawnSet.stream()
            .flatMap { spawnSet: SpawnSet -> spawnSet.spawnInfos.stream() }
            .filter { spawnInfo: SpawnInfo? -> spawnInfo is SpawnInfoPokemon && spawnInfo.pokemonSpec.matches(pokemon) }
            .flatMap { spawnInfo: SpawnInfo -> spawnInfo.condition.biomes.stream() }
            .map { obj: ResourceLocation -> obj.toString() }
            .collect(Collectors.toList())
    }
}